common/src/main/java/io/homeassistant/companion/android/common/data/integration/Entity.kt

```diff
@@ -298,6 +299,50 @@ fun <T> Entity<T>.getLightColor(): Int? {
     }
 }
 
+fun <T> Entity<T>.supportsVolumeSet(): Boolean {
+    return try {
+        if (domain != "media_player") return false
+        ((attributes as Map<*, *>)["supported_features"] as Int) and EntityExt.MEDIA_PLAYER_SUPPORT_VOLUME_SET == EntityExt.MEDIA_PLAYER_SUPPORT_VOLUME_SET
+    } catch (e: Exception) {
+        Log.e(EntityExt.TAG, "Unable to get supportsVolumeSet", e)
+        false
+    }
+}
+
+fun <T> Entity<T>.getVolumeLevel(): EntityPosition? {
+    return try {
+        if (!supportsVolumeSet()) return null
+
+        val minValue = 0f
+        val maxValue = 100f
+
+        // Convert to percentage to match frontend behavior:
+        // https://github.com/home-assistant/frontend/blob/dev/src/dialogs/more-info/controls/more-info-media_player.ts#L137
+        val currentValue = ((attributes as Map<*, *>)["volume_level"] as? Number)?.toFloat()?.times(100) ?: 0f
```

Comment:
Why do `times(100)` if you're going to divide it again when submitting the action and can set the step size?

Reply:
in reality to really read volume level we don't need this method we could just read the attribute and call it a day but to do all the logic needed for device controls display we have this. Even for Wear OS we would want to show the controls as the frontend does and leave the attributes unconverted.

Reply:
The current logic  here is similar to `LightControl` as well where we do some conversion in the helper and the rest in appropriate places (lights for example impacts device controls and wear OS details pane).